packages/core/src/main/start-main-application/lens-window/application-window/session-certificate-verifier.injectable.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import { timingSafeEqual, X509Certificate } from "crypto";
+import type { Request } from "electron";
+import lensProxyCertificateInjectable from "../../../../common/certificate/lens-proxy-certificate.injectable";
+
+// see https://www.electronjs.org/docs/latest/api/session#sessetcertificateverifyprocproc
+export enum ChromiumNetError {
+  SUCCESS = 0,
+  FAILURE = -2,
+  RESULT_FROM_CHROMIUM = -3,
+}
+
+export type CertificateVerificationCallback = (error: ChromiumNetError) => void;
+
+const sessionCertificateVerifierInjectable = getInjectable({
+  id: "session-certificate-verifier",
+  instantiate: (di) => {
+    const lensProxyCertificate = di.inject(lensProxyCertificateInjectable).get();
+    const lensProxyX509Cert = new X509Certificate(lensProxyCertificate.cert);
+
+    return (request: Request, shouldBeTrusted: CertificateVerificationCallback) => {
```

Comment:
This function could be simpler by being named `requestCanBeTrustedInjectable`, and returning `boolean` instead of bleeding abstraction of `ChromiumNetError`. That would mean that call to "shouldBeTrusted" happens somewhere else, and we would have dead-simple code such as:
```
const thereIsTrust = requestCanBeTrusted(request);
shouldBeTrusted(thereIsTrust);
```

Reply:
I disagree. `boolean` is not a good fit because the return type either `YES` or `MAYBE`.